app/controller/button/DigitizeButtonController.js

```diff
@@ -0,0 +1,479 @@
+            radioGroupItems.push(me.getRadioGroupItem(0, true));
+        } else {
+            for (var i = 0; i <= me.contextMenuGroupsCounter; i++) {
+                radioGroupItems.push(me.getRadioGroupItem(i, me.activeGroupIdx === i));
+            }
+        }
+        var menu = Ext.create('Ext.menu.Menu', {
+            width: 100,
+            plain: true,
+            renderTo: Ext.getBody(),
+            items: [{
+                text: 'Start new Group',
+                handler: function(){
+                    me.contextMenuGroupsCounter++;
+                    me.activeGroupIdx = me.contextMenuGroupsCounter;
+                }
+            }, {
+                text: 'Active Group',
+                menu: {
+                    name: 'active-group-submenu',
+                    items: [{
+                        xtype: 'radiogroup',
+                        columns: 1,
+                        vertical: true,
+                        items: radioGroupItems,
+                        listeners: {
+                            'change': function(radioGroup, newVal){
+                                me.activeGroupIdx = newVal.radiobutton;
+                            }
+                        }
+                    }]
+                }
+            }]
+        });
+        menu.showAt(evt.x, evt.y);
+    },
+
+    /**
+     * Handles the draw end event by getting the features and passing them
+     *   to the `prepareRequestParams` function
+     * @param {DrawEvent} evt The openlayers draw event containing the features
+     */
+    handleDrawEnd: function(evt) {
+        // evt.feature if filled after drawend, only contains current finished feature.
+        // evt.features is set on modifyend and will contain all the current features of that geom-type
+        var feat = evt.feature ? evt.feature : evt.features.getArray();
+        this.prepareRequestParams(feat);
+    },
+
+    /**
+     * Prepares the API request with the given features and calls the `doAjaxRequest` method.
+     * If you need custom handling, just overwrite this method.
+     * @param {array[ol.Feature]|ol.Feature} feat The feature or array of features
+     *   that should be used in the request
+     */
+    prepareRequestParams: function(feat) {
+        var me = this;
+        var view = me.getView();
+        var type = view.getType();
+        var format = new ol.format.GeoJSON({
+            dataProjection: me.map.getView().getProjection().getCode()
+        });
+        var searchParams;
+        var jsonParams;
+        if (type === 'Point') {
+            var feats = me.resultLayer.getSource().getFeatures();
+            if (Ext.isEmpty(feats)) {
+                jsonParams = format.writeFeatures([]);
+            } else {
+                jsonParams = format.writeFeatures(feats.filter(function (f) {
+                    return f.getGeometry().getType() === 'Point' &&
+                      f.get('group') === me.activeGroupIdx;
```

Comment:
Uff, I guess this will be hard to read / understand in a couple of months :wink: